cfme/automate/buttons.py

```diff
@@ -853,6 +855,16 @@ def delete_all(self):
         for btn_grp in self.appliance.rest_api.collections.custom_button_sets.all:
             btn_grp.action.delete()
 
+    def instantiate_with_rest(self, id):
```

Comment:
I don't think this method should be added in this way, just to provide it for a REST based test.

What this is really providing is kwargs for instantiate via an ID, and I would like to separate that from the instantiation.

Ideally this could be added to BaseEntity, providing a way to instantiate by REST ID lookup, which would be advantageous to `BaseCollection.all` methods.

Consider, for now, creating a `ButtonGroup.kwargs_by_id` classmethod. That would provide the following pattern to fixtures/tests/framework:

```
def test_buttons(group):
    # using your fixture naming from below, but group here is actually a REST ID
    group_collection = appliance.collections.button_groups
    group_kwargs = group_collection.ENTITY.kwargs_by_id(group)
    group_entity = group_collection.instantiate(group_kwargs)
    assert group_entity.exists  # check it in the UI
    

```

Reply:
Interesting... let me try this. @mshriver I will ping; if found some problems with Entities.

Reply:
@mshriver I like this idea... but added small changes like as we are writing `classmethod` better not return only kawars but instance. also I as you mentioned `all` method use case; I tried to cover it.